Reorganize the folders & sync with coderbyte

ykim-gdl opened the pull request:
The new file structure:

problem/
  - question1/
  - question2/

solutoin/
  - question1/
  - question2/

nealcheng-guideline commented on problem/question2/main.py:
```diff
@@ -43,31 +46,20 @@ def construct_prompt(participants: list[Participant]) -> str:
+    # 1. Load data from "participant_data.json"
 
-    # 2. Fix this to initialize a list of Participant objects
+    # 2. Initialize the Participant objects ensuring that the correct type of "salary_dollars" is used 
```
I'm not sure if I want to give the answer away.  I would much rather prefer them to figure it out.  Can you write
```
Please load the data into a list of Participant and debug any issues.  
```

nealcheng-guideline commented on problem/question2/main.py:
```diff
@@ -43,31 +46,20 @@ def construct_prompt(participants: list[Participant]) -> str:
     # participants = [Participant(**participant) for participant in data["participants"]] 
 
-    # 3. Construct the prompt
+    # 3. Construct the prompt (complete construct_prompt)
```
This should be very obvious and not necessary.

nealcheng-guideline commented on problem/question2/main.py:
```diff
@@ -43,31 +46,20 @@ def construct_prompt(participants: list[Participant]) -> str:
-
-
-
-"""
```
I think we've decided that nobody can get to this extra credit.  I'll just keep this extra credit problem in the back of my mind in case we meet a 10x coder :P.

nealcheng-guideline commented on solution/question2/models.py:
```diff
@@ -16,11 +16,4 @@ def model_dump(self, *args, **kwargs):
         model_dump['salary_dollars'] = str(model_dump['salary_dollars'])
-        return model_dump
-
-    # this is optional and an extra credit.  If can't do this, please 
```
We should keep this part in `solutions` to remind us this is extra credit

nealcheng-guideline commented on solution/question2/question1.py:
```diff
@@ -0,0 +1,30 @@
+                    return Decimal(dollars)
+                except:
+                    pass
+            ###########################
```
I think we can get rid of this `###########################` line.

nealcheng-guideline commented on problem/question2/main.py:
```diff
@@ -17,23 +17,26 @@
 from gemini_client import GeminiClient
 from validation import validate_prompt_input
 
+# This loads env variables
```
I like this

ykim-gdl commented:
Hi @nealcheng-guideline, I’ve addressed most of the comments. Re the rephrasing of problems in Q2, could you include that in your PR since there are already some fixes we need to make to Q2?
I’ll rebase the other PR and incorporate the changes for Q1.
